config/webpack/getThemeWebpackPromise.js

```diff
@@ -0,0 +1,120 @@
+ */
+const themeConfig = (rootPath, themeFile, cachedObject) => (
+  {
+    mode: 'production',
+    entry: {
+      theme: themeFile,
+    },
+    module: {
+      rules: [
+        {
+          test: /\.(scss|css)$/,
+          use: [
+            {
+              loader: MiniCssExtractPlugin.loader,
+            },
+            {
+              loader: 'css-loader',
+              options: {
+                modules: 'global',
+                importLoaders: 2,
+                localIdentName: '[name]__[local]___[hash:base64:5]',
+              },
+            },
+            {
+              loader: 'postcss-loader',
+              options: {
+                // Add unique ident to prevent the loader from searching for a postcss.config file. See: https://github.com/postcss/postcss-loader#plugins
+                ident: 'postcss',
+                plugins() {
+                  return [
+                    rtl(),
+                    Autoprefixer(),
+                  ];
+                },
+              },
+            },
+            {
+              loader: 'sass-loader',
+            },
+          ],
+        }],
+    },
+    plugins: [
+      new MiniCssExtractPlugin({
+        filename: 'temp-themes.css',
+        ignoreOrder: true,
+      }),
+      new PostCSSAssetsPlugin({
+        test: /\.css$/,
+        log: false,
+        plugins: [
+          PostCSSCustomProperties({
+            preserve: true,
+            // Here is where the cachedObject is populated
+            exportTo: [
+              cachedObject,
+            ],
+          }),
+        ],
+      }),
+    ],
+    resolve: {
+      extensions: ['.js'],
+    },
+    resolveLoader: {
+      modules: [path.resolve(path.join(rootPath, 'node_modules'))],
+    },
+  }
+);
+
+/**
+ * Gets a promise that performs webpack on the theme file. The promise resolves with an object that contains the
+ * themeable variables and values.
+ * @param {String} rootPath the root path from where the webpack is being run
+ * @param {String} themeFile the file to be webpacked
+ */
+module.exports = (rootPath, themeFile) => {
+  const cachedObject = {
+    // Provide our own toJSON as the default inserts commas in between things like inset values which is invalid CSS
+    toJSON: customProperties => (
+      Object.keys(customProperties).reduce((customPropertiesJSON, key) => {
+        const aggregatedCustomPropertiesJSON = customPropertiesJSON;
+        aggregatedCustomPropertiesJSON[key] = customProperties[key].map(customPropertyComponent => String(customPropertyComponent)).join(' ');
+        return aggregatedCustomPropertiesJSON;
+      }, {})
+    ),
+    customProperties: {},
+  };
+  return new Promise((resolve, reject) => {
+    const compiler = webpack(themeConfig(rootPath, themeFile, cachedObject));
+    // Set the output file system to MemoryFS so that this all runs in memory
+    compiler.outputFileSystem = new MemoryFS();
```

Comment:
Do we need to run this in memory?

Reply:
I don't want to have anything generated.  Otherwise a new js and css file will be generated.

Reply:
and MemoryFS is already a transitive dependency due to webpack.